catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -0,0 +1,699 @@
+            } catch (InvalidSyntaxException | IOException e) {
+                LOGGER.warn("Error getting registry info: {}", e.getMessage());
+            }
+        }
+
+        Collections.sort(metatypes,
+                (o1, o2) -> ((String) o1.get("id")).compareToIgnoreCase((String) o2.get("id")));
+        return metatypes;
+    }
+
+    @Override
+    public Map<String, Object> allRegistryMetacards() {
+        Map<String, Object> nodes = new HashMap<>();
+        List<Map<String, Object>> registryMetacardInfo = new ArrayList<>();
+
+        try {
+            List<RegistryPackageType> registryMetacardObjects =
+                    federationAdminService.getRegistryObjects();
+
+            List<Metacard> metacards = federationAdminService.getRegistryMetacards();
+            Map<String, Metacard> metacardByRegistryIdMap = getRegistryIdMetacardMap(metacards);
+
+            registryMetacardInfo = getWebMapsFromRegistryPackages(registryMetacardObjects,
+                    metacardByRegistryIdMap);
+
+        } catch (FederationAdminException e) {
+            LOGGER.warn("Couldn't get remote registry metacards '{}'", e);
+        }
+
+        if (customSlots != null) {
+            nodes.put(CUSTOM_SLOTS_KEY, customSlots);
+        }
+
+        Map<String, Object> autoPopulateMap = new HashMap<>();
+        autoPopulateMap.put(SERVICE_BINDINGS_KEY, endpointMap.values());
+        nodes.put(AUTO_POPULATE_VALUES_KEY, autoPopulateMap);
+
+        nodes.put(NODES_KEY, registryMetacardInfo);
+
+        return nodes;
+    }
+
+    private List<Map<String, Object>> getWebMapsFromRegistryPackages(
+            List<RegistryPackageType> packages, Map<String, Metacard> metacardByRegistryIdMap) {
+        List<Map<String, Object>> registryMaps = new ArrayList<>();
+        for (RegistryPackageType registryPackage : packages) {
+            Map<String, Object> registryWebMap = registryMapConverter.convert(registryPackage);
+
+            Metacard metacard = metacardByRegistryIdMap.get(registryPackage.getId());
+            Map<String, Object> transientValues = getTransientValuesMap(metacard);
+            if (MapUtils.isNotEmpty(transientValues)) {
+                registryWebMap.put(TRANSIENT_VALUES_KEY, transientValues);
+            }
+
+            if (MapUtils.isNotEmpty(registryWebMap)) {
+                registryMaps.add(registryWebMap);
+            }
+        }
+        return registryMaps;
+    }
+
+    private Metacard getRegistryMetacardFromRegistryPackage(RegistryPackageType registryPackage)
+            throws FederationAdminException {
+        if (registryPackage == null) {
+            throw new FederationAdminException(
+                    "Error creating metacard from registry package. Null package was received.");
+        }
+        Metacard metacard;
+
+        try {
+            JAXBElement<RegistryPackageType> jaxbRegistryObjectType =
+                    EbrimConstants.RIM_FACTORY.createRegistryPackage(registryPackage);
+
```

Comment:
The tools will still complain unless the close happens in a finally or you use try with resources. If an exception occurred in the block, it would skip the close (which of course does nothing).